eslint.config.mjs

```diff
@@ -0,0 +1,278 @@
+          leadingUnderscore: 'allow',
+          trailingUnderscore: 'forbid',
+        },
+        {
+          selector: 'enumMember',
+          format: ['PascalCase'],
+        },
+        {
+          selector: 'import',
+          format: ['camelCase', 'PascalCase', 'snake_case', 'UPPER_CASE'],
+        },
+        {
+          selector: 'interface',
+          format: ['PascalCase'],
+          custom: {
+            regex: '^I[A-Z]',
+            match: false,
+          },
+        },
+        {
+          selector: 'objectLiteralMethod',
+          format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+        },
+        {
+          selector: 'objectLiteralProperty',
+          format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+        },
+        {
+          selector: 'typeLike',
+          format: ['PascalCase'],
+        },
+        {
+          selector: 'typeParameter',
+          format: ['PascalCase'],
+          custom: {
+            regex: '^.{3,}',
+            match: true,
+          },
+        },
+        {
+          selector: 'variable',
+          format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+          leadingUnderscore: 'allow',
+        },
+        {
+          selector: 'parameter',
+          format: ['camelCase', 'PascalCase'],
+          leadingUnderscore: 'allow',
+        },
+        {
+          selector: [
+            'classProperty',
+            'objectLiteralProperty',
+            'typeProperty',
+            'classMethod',
+            'objectLiteralMethod',
+            'typeMethod',
+            'accessor',
+            'enumMember',
+          ],
+          format: null,
+          modifiers: ['requiresQuotes'],
+        },
+      ],
+    },
+  },
+
+  // Node.js + TypeScript scripts
+  {
+    files: [
+      '**/scripts/**/*.ts',
+      '**/scripts/**/*.mts',
+      'packages/snaps-execution-environments/scripts/**/*.ts',
+    ],
+
+    extends: nodejs,
+
+    languageOptions: {
+      sourceType: 'module',
+      parserOptions: {
+        tsconfigRootDir: import.meta.dirname,
+        project: ['./tsconfig.json'],
+      },
+    },
+
+    rules: {
+      'n/hashbang': 'off',
+    },
+  },
+
+  // CommonJS Node.js scripts
+  {
+    files: ['**/*.js', '**/*.cjs'],
+    extends: nodejs,
+
+    languageOptions: {
+      sourceType: 'script',
+    },
+
+    rules: {
+      'n/hashbang': 'off',
+    },
+  },
+
+  // ESM Node.js scripts
+  {
+    files: ['**/*.mjs'],
+    extends: nodejs,
+
+    languageOptions: {
+      sourceType: 'module',
+    },
+
+    rules: {
+      'n/hashbang': 'off',
+    },
+  },
+
+  // Test files
+  {
+    files: [
+      '**/*.test.ts',
+      '**/*.test.tsx',
+      '**/*.test.browser.ts',
+      '**/*.test.js',
+    ],
+    extends: [jest, nodejs],
+
+    rules: {
+      // This rule is too strict for test files.
+      // TODO: Upstream this change to `@metamask/eslint-config-jest`.
+      '@typescript-eslint/unbound-method': 'off',
```

Comment:
I think we can remove some comments in some of the controllers with this change

Reply:
Done in #3112. I split them up to keep it reviewable.